src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -351,6 +350,7 @@ protected TypeDesc GetDefaultNativeType(NativeType nativeType)
 
                 case MarshallerKind.UnicodeString:
                     return GetDefaultNativeType(NativeType.SysInt); // We pin UniCodeString, so native type will be SysInt
```

Comment:
The extra level of indirection is not buying us much, so got rid of GetDefaultNativeType.